### ISSUE TITLE
CollectionView.SelectionChanged gets fired if a CollectionView.SwipeView is added to ItemTemplate [ANDROID]

### ISSUE THREAD

**Issue by Lazareena** [dotnet/maui#15778]
### Description

If a CollectionView.SwipeItem is added to the ItemTemplate of the CollectionView, the swipe gesture also triggers the SelectionChanged handler

### Steps to Reproduce

1. Add a MAUI CollectionView to your application.
2. Set SelectionMode to Single.
3. Add a SelectionChanged event handler.
4. Add a SwipeView to the ItemTemplate and add a SwipeItem to the SwipeView.LeftItems collection.
5. Run the application and put a breakpoint in the SelectionChanged event handler. The event handler will get hit when swiping left on an item.

### Link to public reproduction project repository

https://github.com/Lazareena/CollectionViewSelectedItemBug

### Version with bug

7.0.49

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

N/A

**Comment by Srki94** [comment]
I've just run into this issue while trying to  port my Xamarin Forms application to MAUI.
Honestly I've run into more issues regarding basic functionality in MAUI than I ever had in Xamarin Forms to  begin  with and the reason  why I wanted to switch to MAUI is to avoid basic functionality issues ...

Are there any workaround suggestions?

I need to show another page with data when user taps an entry in a collection view, but I also have swipe view on every item in the list which exposes Edit and Remove buttons. 

P.S. ListView is no good, it has issues with SwipeView also - not closing them and removing SwipeView content.

**Comment by FM1973** [comment]
I can confirm this. Even in 7.0.92.
On Ios the behavior is like expected. On Android not.

What you can do is not to use the SelectionChangedCommand / Event of the CollectionView.
Instead implement a Tap-Handler inside the DataTemplate.

**Comment by Morvayn** [comment]
Hi there
Check this out as a work arround
https://stackoverflow.com/questions/76645317/net-maui-prevent-swipe-from-triggering-parent-collectionview-selected-item/76660641#76660641

**Comment by kevinxufei** [comment]
Verified this issue with Visual Studio Enterprise 17.9.0 Preview 4, swiping an Item on an Android device triggers the event and hits the breakpoint.

**Comment by Quaybe** [comment]
Confirming this behavior. Please fix!

**Comment by ThisIsDavidHan** [comment]
Confirming this behavior in VS Version 17.10.4 and Microsoft.Maui.Controls Version 8.0.70.

Our workaround consists of these additions:

**XAML page:**
``` XAML
<ContentPage.Resources>
    <ResourceDictionary>
        <TapGestureRecognizer x:Key="EnableSelection" Tapped="EnableSelection" />
    </ResourceDictionary>
</ContentPage.Resources>

<CollectionView.ItemTemplate>
    <DataTemplate>
        <SwipeView SwipeStarted="DisableSelection">
            <Grid>
                <Grid.GestureRecognizers>
                    <OnPlatform x:TypeArguments="GestureRecognizer">
                        <On Platform="Android" Value="{StaticResource EnableSelection}" />
                    </OnPlatform>
                </Grid.GestureRecognizers>
                ...
            </Grid>
        </SwipeView>
    </DataTemplate>
</CollectionView.ItemTemplate>
```

**Code behind:**
``` C#
protected override void OnAppearing()
{
    base.OnAppearing();

#if ANDROID
    ViewModel.EnableSelection+= EnableSelection;
#endif
}

#if ANDROID
protected override void OnDisappearing()
{
    base.OnDisappearing();

    ViewModel.EnableSelection -= EnableSelection;
}
#endif

private void DisableSelection(object sender, SwipeStartedEventArgs e)
{
#if ANDROID
    MyCollectionView.SelectionMode = SelectionMode.None;
#endif
}

private void EnableSelection(object? sender, EventArgs? e)
{
#if ANDROID
    MyCollectionView.SelectionMode = SelectionMode.Single;
#endif
}
```

**ViewModel:**
``` C#

#if ANDROID
    public event EventHandler? EnableSelection;
#endif

<!-- On relevant place where delete item is handled -->
#if ANDROID
        EnableSelection?.Invoke(this, EventArgs.Empty);
#endif
```

We are looking forward to this fix as our workaround adds a lot of clutter to the code :)

**Comment by josuejs23** [comment]
> ```
> <ContentPage.Resources>
>     <ResourceDictionary>
>         <TapGestureRecognizer x:Key="EnableSelection" Tapped="EnableSelection" />
>     </ResourceDictionary>
> </ContentPage.Resources>
> ```

Cloud you give more explanation about your solution, this code gives me this exception "Specified cast is not valid." for the SwipeView.

**Comment by ThisIsDavidHan** [comment]
From the time I posted our solution we have changed our workaround a bit and it might help you.
We had som issues in Release builds on iOS (Debug was OK). 
We no longer use Resources.


This is updated XAML:
``` XAML
<CollectionView.ItemTemplate>
    <DataTemplate x:DataType="x:String">
        <SwipeView SwipeStarted="DisableSelection">
            <SwipeView.RightItems>
                <SwipeItem ... />
            </SwipeView.RightItems>
            <OnPlatform x:TypeArguments="Grid">
                <OnPlatform.Platforms>
                    <On Platform="Android">
                        <Grid>
                            <Grid.GestureRecognizers>
                                <TapGestureRecognizer Tapped="EnableSelection"/>
                            </Grid.GestureRecognizers>
                            ...
                        </Grid>
                    </On>
                    <On Platform="iOS">
                        <Grid>
                            ...
                        </Grid>
                    </On>
                </OnPlatform.Platforms>
            </OnPlatform>
        </SwipeView>
    </DataTemplate>
</CollectionView.ItemTemplate>
```

**Comment by JkPersson** [comment]
Please prioritize this.

**Comment by TTriisa** [comment]
The workaround of setting SelectionMode to None on SwipeStarted doesn't work with SelectionMode Multiple, since setting it to None clears the selection for all items. This makes the behavior of using SwipeView within multi-select CollectionView quite unpleasant for android. Please prioritize.

**Comment by ThisIsDavidHan** [comment]
> The workaround of setting SelectionMode to None on SwipeStarted doesn't work with SelectionMode Multiple, since setting it to None clears the selection for all items. This makes the behavior of using SwipeView within multi-select CollectionView quite unpleasant for android. Please prioritize.

You’re right — as shown in my [previous comment](https://github.com/dotnet/maui/issues/15778#issuecomment-2245677806), we were using SelectionMode.Single, which is why the workaround worked in our case. However, when using SelectionMode.Multiple, setting it to None in SwipeStarted clears all selected items, making SwipeView within a multi-select CollectionView quite frustrating on Android. I agree, it would be great if this could be prioritized.

**Comment by burrowj** [comment]
Maybe we could have a setting for what triggers selection.  Eg LongPress, SingleTap etc.  This may diffrenetiate between a swipe and a tap

**Comment by burrowj** [comment]
Just to add.  I was going to disable selection until the swipeview is closed.  However the SwipeEnded is not called when the swipeview closes and there is no close event that I can see.  Makes it hard to work around this

**Comment by marciosouzajunior** [comment]
I worked around it by setting SelectionMode="None" and using a TapGestureRecognizer on the row. This avoids the swipe/tap conflict on Android and works fine for me.
